crates/bevy_ecs/src/query/state.rs

```diff
@@ -83,6 +87,7 @@ pub struct QueryState<D: QueryData, F: QueryFilter = ()> {
     pub(super) is_dense: bool,
     pub(crate) fetch_state: D::State,
     pub(crate) filter_state: F::State,
+    sync_by_observer: bool,
```

Comment:
This now can be moved to `QueryState` wrapper but then we still have to duplicate a lot of QueryState's methods